### DISCUSSION TITLE
simple random number generation is taking a lot of time.

### DISCUSSION THREAD

**Discussion by Aaditya-23** [prisma/prisma#18656] [Q&A] [answered]
**I am not comparing prisma with mongoose, but using it as a reference here**

[app1](https://remix-notifier.vercel.app)
here I am using remix, prisma, planetscale. when You click on the add button to add a notification, it takes about 1.5 seconds to get the response back. 

 
![app1](https://user-images.githubusercontent.com/74983108/230441700-eadc0eb1-6f5d-41a2-a9a6-918b9cb96a89.png)

The function is simply generating 3 random numbers and returning them.

[app2](https://remix-mongo-nu.vercel.app)
here I am doing the same thing, but instead using mongoose and it takes about 200ms to get the response back.

![app2](https://user-images.githubusercontent.com/74983108/230441783-c9662ee8-9d36-4203-aba6-c70296644e54.png)

In my other apps where I am using prisma, requests take up a lot of time.

**Comment by ludralph** [comment]
Hi @Aaditya-23 👋 

Please enable [tracing](https://www.prisma.io/docs/concepts/components/prisma-client/opentelemetry-tracing) to understand the time to execute your query and analyze the application performance and identify bottlenecks. It is also worth pointing out that Prisma and Mongoose have different implementations and may have different performance characteristics.

**Reply by Aaditya-23** [reply]
hi @ludralph 

I looked up the Prisma docs, and they say that I need to set up connection pooling if I am using serverless functions (which I am using). But I cannot figure out how to do it with planetscale. Can you help me a little bit with this?

**Comment by Aaditya-23** [comment, marked as answer]
sorry, my vercel app and db  was hosted in different regions which caused this latency.
